### PR TITLE
Test vsam stcs tabs

### PR DESCRIPTION
**Pull request by Himani1519**
## Proposed changes

This PR addresses Issue: [*Link to Github issue within https://github.com/zowe/zen/issues* if any]

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets a "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zen/blob/v2.x/staging/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made

- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

File: playwright_test/Tests/Stcs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		config.SECURITY_STC_ZIS
+	  )
+    await stcsPage.movetoStcsPage()
+  })
+
+  test.afterEach(async () => {
+    await electronApp.close()
+  })
+
+   test('test title of page', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: this test and "test all required fields' Should be one test. Not necessary to constantly run this every time.

File: playwright_test/Tests/Stcs.spec.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+   })
+
+   test('test all required fields', async ({ page }) => {
+    await expect(stcsPage.zis).toBeTruthy()
+    await expect(stcsPage.zowe).toBeTruthy()
+    await expect(stcsPage.aux).toBeTruthy()
+    await expect(stcsPage.dataset_proclib).toBeTruthy()
+    })
+
+   test('verify yaml on zos', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: what is the purpose of this test?

File: playwright_test/Tests/Stcs.spec.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    await electronApp.close()
+  })
+
+   test('test title of page', async ({ page }) => {
+   const title = await stcsPage.returnTitleOfStcsPage();
+   expect(title).toBe(STCS_TITLE);
+   })
+
+   test('test all required fields', async ({ page }) => {
+    await expect(stcsPage.zis).toBeTruthy()
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: you should also check if the value match what we have passed from previous steps....

File: playwright_test/Tests/Stcs.spec.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    await stcsPage.closeButton()
+    })
+
+   test('Test view job', async ({ page }) => {
+     await stcsPage.click_previewJob()
+     await expect(stcsPage.editor_title_element).toBeTruthy()
+     await stcsPage.closeButton()
+    })
+
+   test('click Previous step button', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: this test and 'Test previous button is enabled' should be one test

File: playwright_test/Tests/Stcs.spec.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+     const title = await stcsPage.returnTitleOfPrevPage();
+     expect(title).toBe(SECURITY_TITLE);
+   })
+   test('Test previous button is enabled', async ({ page }) => {
+     const is_prevButtonEnable = await stcsPage.isPreviousButtonEnable();
+     expect(is_prevButtonEnable).toBe(true);
+    })
+
+   test('Test Skip Stcs button is enable', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts:160]: this test and 'test click skip STCS button' should be one test

File: playwright_test/Tests/Stcs.spec.ts [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+     const DatasetProclib_Value = await stcsPage.get_datasetProclib_value();
+     expect(ZoweValue).toBe(SECURITY_STC_ZOWE);
+     expect(Zis_Value).toBe(SECURITY_STC_ZIS);
+     expect(Aux_Value).toBe(SECURITY_AUX);
+     expect(DatasetProclib_Value).toBe(PROC_LIB);
+     })
+
+   test('verify stcs applied successfully on zos', async ({ page }) => {
+      await stcsPage.initializeSTC()
+      const command = `tsocmd "LISTDS '${config.PROC_LIB}' MEMBERS"`;
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: well this test might pass for now, but you might need to delete them first then do this to actually ensure it worked..

File: playwright_test/Tests/Stcs.spec.ts [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+          const yaml = await runSSHCommand(command);
+          expect(yaml).toContain(SECURITY_AUX);
+          expect(yaml).toContain(SECURITY_STC_ZOWE);
+          expect(yaml).toContain(SECURITY_STC_ZIS);
+      } catch (error) {
+          console.error('Error executing command:', error.message);
+      }
+      });
+
+   test('Test initialize stcs', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: the previous test does all of this....

File: playwright_test/Tests/Stcs.spec.ts [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+     expect(isWriteConfig_check_visible).toBe(true);
+     const isUploadConfig_check_visible = await stcsPage.isUploadConfigGreenCheckVisible();
+     expect(isUploadConfig_check_visible).toBe(true);
+     const isInitStcs_check_visible = await stcsPage.isInitSTCSGreenCheckVisible();
+     expect(isInitStcs_check_visible).toBe(true);
+     const is_GreenCheck_Visible = await stcsPage.isStatusChecked();
+     expect(is_GreenCheck_Visible).toBe(false);
+     })
+
+   test('Test continue Button enabled after init stcs', async ({ page }) => {
```
Comment (struga0258) [inline playwright_test/Tests/Stcs.spec.ts]: this should be tested with the top two test...

File: playwright_test/Tests/Stcs.spec.ts (hunk 2 above)
Comment (Himani1519) [reply playwright_test/Tests/Stcs.spec.ts]: It checks whether yaml also updated on zos or not

File: playwright_test/Tests/Stcs.spec.ts (hunk 6 above)
Comment (Himani1519) [reply playwright_test/Tests/Stcs.spec.ts]: Yes, for that I will create a separate ticket for cleanup code.

**Comment by struga0258** [comment]
The new code changes pushed in this PR break existing test, this PR is not ready for merge and needs to be fixed.

1. You have replaced setup.js with setup.ts not sure why, but then you have renamed some of the functions which are now causing errors. 
2. We had spoke about this before but going from zos-node-accessor to node-ssh might be the easy solution but it comes with limitations that will not help us debug and test Zen properly. 
3. You have added two new test, Stcs and Vsam, but they are both missing 'prepare_enviroment'. This is probably why you didn't get the errors from setup.ts.
4. The test also need to be improved for the new test files you added, they can be improved and test better.

**Review by struga0258 — APPROVED** [review]

**Pull request merged**